app/src/main/java/com/github/versus/LocationFragment.java

```diff
@@ -253,18 +268,81 @@ public boolean onOptionsItemSelected(MenuItem item) {
             openPlacesDialog();
 
         }
+        else if(item.getItemId() == R.id.option_choose_location){
+            enableCustomLocationSelection();
+        }
 
 
 
 
         return true;
     }
 
+    /**
+     * Enables custom location selection by setting a map click listener.
+     * When the map is clicked, it calls the findClosestFields method to find the closest fields to the clicked position.
+     */
+    private void enableCustomLocationSelection() {
+        map.setOnMapClickListener(new GoogleMap.OnMapClickListener() {
+            @Override
+            public void onMapClick(LatLng latLng) {
+                // Find closest fields to the clicked position
+                findClosestFields(latLng);
+            }
+        });
+    }
 
     /**
-     * Gets the current location of the device, and positions the map's camera.
+     * Finds the closest fields to a clicked position on the map.
+     * Iterates through the custom places and finds the ones within the threshold distance.
+     * Prepares the data for the showPlacesList() method and shows the list of nearby fields.
+     *
+     * @param clickedPosition LatLng object representing the position where the user clicked on the map.
      */
+    private void findClosestFields(LatLng clickedPosition) {
+        double thresholdDistance = 1000.0; // Threshold distance to consider a field as nearby (in meters)
+
+        // Remove the last clicked marker if it exists
+        if (lastClickedMarker != null) {
+            lastClickedMarker.remove();
+        }
+
+        // Add a new marker to the map at the clicked position
+        lastClickedMarker = map.addMarker(new MarkerOptions()
+                .position(clickedPosition)
+                .icon(BitmapDescriptorFactory.defaultMarker(BitmapDescriptorFactory.HUE_AZURE))
+                .title("Clicked Location"));
+
+        // Iterate through the custom places and find the ones within the threshold distance
+        List<CustomPlace> nearbyFields = new ArrayList<>();
+        for (CustomPlace customPlace : customPlaces) {

```

Comment:
Maybe that's just my obsession with functional programming, but isn't it better to use [streams](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/stream/Stream.html) ?

Reply:
Nice , it seems more concise this way , lets do this!